modules/cxf/src/test/java/org/mule/module/cxf/issues/CxfClientRaceConditionTestCase.java

```diff
@@ -99,10 +133,30 @@ private void cxfJmsTest(String inboundQueue, String outboundQueue, String expect
         consumerConnection.close();
     }
 
+    private void cxfHTTPTest(String protocol, String host, Integer port, String method, HttpRequestOptions options) throws MuleException
+    {
+        long errorCount = 0;
+
+        for (int i = 0; i < totalMessages; i++)
+        {
+
+            MuleMessage request = new DefaultMuleMessage(HTTP_REQUEST_BODY, muleContext);
+            MuleClient client = muleContext.getClient();
+            MuleMessage message = client.send(protocol + "://" + host + ":" + port + "/" + method, request, options);
+            String response = IOUtils.toString((PipedInputStream) message.getPayload());
+            if (EXPECTED_HTTP_MESSAGE.length() != response.length())
```

Comment:
why not assert directly here? this kind of test failures are hard to troubleshoot